BMS Workspace-git/bms-1227-fw/Phantom_drivers/rtos/source/task_command_line.c

```diff
@@ -0,0 +1,43 @@
+/*
+ * task_command_line.c
+ *
+ *  Created on: Oct 25, 2022
+ *      Author: lonam
+ */
+#include "task_command_line.h"
+
+#include "FreeRTOS.h"
+#include "FreeRTOSConfig.h"
+#include "os_task.h"
+
+#include "Phantom_sci.h"
+#include "hwConfig.h"
+
+static uint8 previous;
+static xTaskHandle commandLineTaskHandle;
+static void CommandLineTask(void* params){
+    while(1){
+        uint8 msg;
+        sciReceive(PC_UART, 1, &msg);
+        if(msg==previous){
+            // do nothing
+        } else if(msg=='1'){
+            UARTprintf("Message 1 received\r\n");
+        } else if(msg=='2'){
+            UARTprintf("Message 2 received\r\n");
+        } else if(msg=='3'){
+            UARTprintf("Message 3 received\r\n");
+        }
+        previous = msg;
+        vTaskDelay(1000);
+    }
+}
```

Comment:
it's easy to see now that there are 3 logical parts:
1. wait for a message
2. dispatch/process message
3. update variables

```
// wait for user input
uint8 msg;
sciReceive();

// process input
if ()
{
}

// update previous
previous = msg;

vTaskDelay(1000);
```
 spacing (along with comments) can make this clearer